src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/review/ReviewEntityMapper.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.review
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.CompletedReview
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewConductedBy
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.note.NoteEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.review.ReviewEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.note.NoteMapper
+
+@Component
+class ReviewEntityMapper {
+
+  fun fromEntityToDomain(reviewEntity: ReviewEntity, reviewNoteEntity: NoteEntity): CompletedReview =
+    with(reviewEntity) {
+      CompletedReview(
+        reference = reference,
+        prisonNumber = prisonNumber,
+        deadlineDate = deadlineDate,
+        completedDate = completedDate,
+        note = NoteMapper.fromEntityToDomain(reviewNoteEntity),
+        createdBy = createdBy!!,
+        createdAt = createdAt!!,
+        createdAtPrison = createdAtPrison,
+        conductedBy = toReviewConductedBy(this),
+      )
+    }
+
+  private fun toReviewConductedBy(reviewEntity: ReviewEntity): ReviewConductedBy? =
+    if (reviewEntity.conductedBy != null && reviewEntity.conductedByRole != null) {
+      ReviewConductedBy(name = reviewEntity.conductedBy, role = reviewEntity.conductedByRole)
+    } else {
+      null
```

Comment:
that else null is a little bit jarring could do: 

private fun toReviewConductedBy(reviewEntity: ReviewEntity): ReviewConductedBy? =
    reviewEntity.takeIf { it.conductedBy != null && it.conductedByRole != null }
        ?.let { ReviewConductedBy(name = it.conductedBy!!, role = it.conductedByRole!!) }

Reply:
Yeah, much more "kotlin" - great suggestion, will do that now ... 👍

Reply:
Done 👍